src/Backends/WaylandBackend.cpp

```diff
@@ -1172,7 +1200,28 @@ namespace gamescope
 
     void CWaylandConnector::SetSelection( std::shared_ptr<std::string> szContents, GamescopeSelection eSelection )
     {
-        // Do nothing
+        if ( m_pBackend->m_pDataDeviceManager && !m_pBackend->m_pDataDevice )
+            m_pBackend->m_pDataDevice = wl_data_device_manager_get_data_device( m_pBackend->m_pDataDeviceManager, m_pBackend->m_pSeat );
+
+        if ( m_pBackend->m_pPrimarySelectionDeviceManager && !m_pBackend->m_pPrimarySelectionDevice )
+            m_pBackend->m_pPrimarySelectionDevice = zwp_primary_selection_device_manager_v1_get_device( m_pBackend->m_pPrimarySelectionDeviceManager, m_pBackend->m_pSeat );
+
+        if ( eSelection == GAMESCOPE_SELECTION_CLIPBOARD && m_pBackend->m_pDataDevice )
+        {
+            m_pBackend->m_pClipboard = szContents;
+            wl_data_source *source = wl_data_device_manager_create_data_source( m_pBackend->m_pDataDeviceManager );
+            wl_data_source_add_listener( source, &m_pBackend->s_DataSourceListener, m_pBackend );
+            wl_data_source_offer( source, "text/plain" );
```

Comment:
I think `text/plain;charset=utf-8` is more compatible than `text/plain`, I know some programs which only look for the former. Though it doesn't hurt to just offer all of `text/plain;charset=utf8`/`text/plain`/`TEXT`/`STRING`/`UTF8_STRING`.